Tablet server failed to load Root tablet during 2.0 upgrade

Found this while trying to upgrade from `1.9.3-SNAPSHOT` to `2.0.0-SNAPSHOT`

```
2019-03-08 15:25:19,166 [tserver.TabletServer] INFO : Root tablet loaded: +r<<
2019-03-08 15:25:19,342 [tserver.FileManager] ERROR: Failed to open file hdfs://localhost:8020/accumulo/tables/+r/root_tablet/A000001b.rf null
2019-03-08 15:25:19,347 [scan.LookupTask] WARN : lookup failed for tablet +r<<
java.io.IOException: Failed to open hdfs://localhost:8020/accumulo/tables/+r/root_tablet/A000001b.rf
        at org.apache.accumulo.tserver.FileManager.reserveReaders(FileManager.java:361)
        at org.apache.accumulo.tserver.FileManager.access$600(FileManager.java:63)
        at org.apache.accumulo.tserver.FileManager$ScanFileManager.openFiles(FileManager.java:526)
        at org.apache.accumulo.tserver.FileManager$ScanFileManager.openFileRefs(FileManager.java:511)
        at org.apache.accumulo.tserver.FileManager$ScanFileManager.openFiles(FileManager.java:536)
        at org.apache.accumulo.tserver.tablet.ScanDataSource.createIterator(ScanDataSource.java:182)
        at org.apache.accumulo.tserver.tablet.ScanDataSource.iterator(ScanDataSource.java:140)
        at org.apache.accumulo.core.iterators.system.SourceSwitchingIterator.seek(SourceSwitchingIterator.java:226)
        at org.apache.accumulo.tserver.tablet.Tablet.lookup(Tablet.java:603)
        at org.apache.accumulo.tserver.tablet.Tablet.lookup(Tablet.java:757)
        at org.apache.accumulo.tserver.scan.LookupTask.run(LookupTask.java:116)
        at org.apache.accumulo.tserver.session.ScanSession$ScanMeasurer.run(ScanSession.java:54)
        at org.apache.accumulo.tserver.TabletServerResourceManager.executeReadAhead(TabletServerResourceManager.java:936)
        at org.apache.accumulo.tserver.TabletServer$ThriftClientHandler.continueMultiScan(TabletServer.java:842)
        at org.apache.accumulo.tserver.TabletServer$ThriftClientHandler.startMultiScan(TabletServer.java:812)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.accumulo.core.trace.TraceUtil.lambda$wrapService$8(TraceUtil.java:235)
        at com.sun.proxy.$Proxy21.startMultiScan(Unknown Source)
        at org.apache.accumulo.core.tabletserver.thrift.TabletClientService$Processor$startMultiScan.getResult(TabletClientService.java:2757)
        at org.apache.accumulo.core.tabletserver.thrift.TabletClientService$Processor$startMultiScan.getResult(TabletClientService.java:2736)
        at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:38)
        at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:39)
        at org.apache.accumulo.server.rpc.TimedProcessor.process(TimedProcessor.java:65)
        at org.apache.thrift.server.AbstractNonblockingServer$FrameBuffer.invoke(AbstractNonblockingServer.java:518)
        at org.apache.accumulo.server.rpc.CustomNonBlockingServer$CustomFrameBuffer.invoke(CustomNonBlockingServer.java:112)
        at org.apache.thrift.server.Invocation.run(Invocation.java:18)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at org.apache.accumulo.fate.util.LoggingRunnable.run(LoggingRunnable.java:35)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
        at java.util.Objects.requireNonNull(Objects.java:203)
        at org.apache.accumulo.core.crypto.CryptoUtils.writeParams(CryptoUtils.java:65)
        at org.apache.accumulo.core.file.rfile.bcfile.BCFile$Reader.serializeMetadata(BCFile.java:585)
        at org.apache.accumulo.core.file.blockfile.impl.CachableBlockFile$Reader$BCFileLoader.load(CachableBlockFile.java:229)
        at org.apache.accumulo.core.file.blockfile.cache.lru.SynchronousLoadingBlockCache.getBlock(SynchronousLoadingBlockCache.java:127)
        at org.apache.accumulo.core.file.blockfile.cache.lru.SynchronousLoadingBlockCache.resolveDependencies(SynchronousLoadingBlockCache.java:64)
        at org.apache.accumulo.core.file.blockfile.cache.lru.SynchronousLoadingBlockCache.getBlock(SynchronousLoadingBlockCache.java:109)
        at org.apache.accumulo.core.file.blockfile.impl.CachableBlockFile$Reader.getMetaBlock(CachableBlockFile.java:385)
        at org.apache.accumulo.core.file.rfile.RFile$Reader.<init>(RFile.java:1156)
        at org.apache.accumulo.core.file.rfile.RFile$Reader.<init>(RFile.java:1251)
        at org.apache.accumulo.core.file.rfile.RFileOperations.getReader(RFileOperations.java:53)
        at org.apache.accumulo.core.file.rfile.RFileOperations.openReader(RFileOperations.java:68)
```

--- a/core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java
+++ b/core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java
@@ -36,6 +36,7 @@ import java.util.TreeMap;
 import org.apache.accumulo.core.crypto.CryptoUtils;
 import org.apache.accumulo.core.cryptoImpl.CryptoEnvironmentImpl;
 import org.apache.accumulo.core.cryptoImpl.NoFileDecrypter;
+import org.apache.accumulo.core.cryptoImpl.NoFileEncrypter;
 import org.apache.accumulo.core.file.rfile.bcfile.Compression.Algorithm;
 import org.apache.accumulo.core.file.rfile.bcfile.Utils.Version;
 import org.apache.accumulo.core.file.streams.BoundedRangeFileInputStream;
@@ -634,6 +635,7 @@ public final class BCFile {
       // backwards compatibility
       if (version.equals(API_VERSION_1)) {
         LOG.trace("Found a version 1 file to read.");
+        decryptionParams = new NoFileEncrypter().getDecryptionParameters();
         this.decrypter = new NoFileDecrypter();
       } else {
         // read crypto parameters and get decrypter

--- a/core/src/test/java/org/apache/accumulo/core/file/rfile/RFileTest.java
+++ b/core/src/test/java/org/apache/accumulo/core/file/rfile/RFileTest.java
@@ -1702,28 +1702,28 @@ public class RFileTest {
 
   @Test(expected = NullPointerException.class)
   public void testMissingUnreleasedVersions() throws Exception {
-    runVersionTest(5, DefaultConfiguration.getInstance());
+    runVersionTest(5, getAccumuloConfig(CryptoTest.CRYPTO_OFF_CONF));
   }
 
   @Test
   public void testOldVersions() throws Exception {
-    AccumuloConfiguration defaultConfiguration = DefaultConfiguration.getInstance();
-    runVersionTest(3, defaultConfiguration);
-    runVersionTest(4, defaultConfiguration);
-    runVersionTest(6, defaultConfiguration);
-    runVersionTest(7, defaultConfiguration);
+    ConfigurationCopy defaultConf = getAccumuloConfig(CryptoTest.CRYPTO_OFF_CONF);
+    runVersionTest(3, defaultConf);
+    runVersionTest(4, defaultConf);
+    runVersionTest(6, defaultConf);
+    runVersionTest(7, defaultConf);
   }
 
   @Test
   public void testOldVersionsWithCrypto() throws Exception {
-    AccumuloConfiguration cryptoOnConf = getAccumuloConfig(CryptoTest.CRYPTO_ON_CONF);
+    ConfigurationCopy cryptoOnConf = getAccumuloConfig(CryptoTest.CRYPTO_ON_CONF);
     runVersionTest(3, cryptoOnConf);
     runVersionTest(4, cryptoOnConf);
     runVersionTest(6, cryptoOnConf);
     runVersionTest(7, cryptoOnConf);
   }
 
-  private void runVersionTest(int version, AccumuloConfiguration aconf) throws IOException {
+  private void runVersionTest(int version, ConfigurationCopy aconf) throws Exception {
     InputStream in = this.getClass().getClassLoader()
         .getResourceAsStream("org/apache/accumulo/core/file/rfile/ver_" + version + ".rf");
     ByteArrayOutputStream baos = new ByteArrayOutputStream();
@@ -1735,8 +1735,15 @@ public class RFileTest {
     byte[] data = baos.toByteArray();
     SeekableByteArrayInputStream bais = new SeekableByteArrayInputStream(data);
     FSDataInputStream in2 = new FSDataInputStream(bais);
+    aconf.set(Property.TSERV_CACHE_MANAGER_IMPL, LruBlockCacheManager.class.getName());
+    aconf.set(Property.TSERV_DEFAULT_BLOCKSIZE, Long.toString(100000));
+    aconf.set(Property.TSERV_DATACACHE_SIZE, Long.toString(100000000));
+    aconf.set(Property.TSERV_INDEXCACHE_SIZE, Long.toString(100000000));
+    BlockCacheManager manager = BlockCacheManagerFactory.getInstance(aconf);
+    manager.start(new BlockCacheConfiguration(aconf));
     CachableBuilder cb = new CachableBuilder().input(in2).length(data.length).conf(hadoopConf)
-        .cryptoService(CryptoServiceFactory.newInstance(aconf, ClassloaderType.JAVA));
+        .cryptoService(CryptoServiceFactory.newInstance(aconf, ClassloaderType.JAVA))
+        .index(manager.getBlockCache(CacheType.INDEX)).data(manager.getBlockCache(CacheType.DATA));
     Reader reader = new RFile.Reader(cb);
     checkIndex(reader);
 
@@ -1779,10 +1786,11 @@ public class RFileTest {
       assertFalse(iter.hasTop());
     }
 
+    manager.stop();
     reader.close();
   }
 
-  public static AccumuloConfiguration getAccumuloConfig(String cryptoOn) {
+  public static ConfigurationCopy getAccumuloConfig(String cryptoOn) {
     ConfigurationCopy cfg = new ConfigurationCopy(DefaultConfiguration.getInstance());
     switch (cryptoOn) {
       case CryptoTest.CRYPTO_ON_CONF:

